src/app/ReadClient.cpp

```diff
@@ -520,6 +520,15 @@ CHIP_ERROR ReadClient::ProcessAttributeReportIBs(TLV::TLVReader & aAttributeRepo
 CHIP_ERROR ReadClient::RefreshLivenessCheckTimer()
 {
     CancelLivenessCheckTimer();
+    uint64_t timeoutMsec = mMaxIntervalCeilingSeconds;
+    uint64_t marginSec   = 0;
+    if (mpExchangeCtx != nullptr && mpExchangeCtx->IsUDPTransport())
+    {
+        marginSec = (CHIP_CONFIG_RMP_DEFAULT_MAX_RETRANS + 1) * mpExchangeCtx->GetIdleRetransmitTimeoutTick();
```

Comment:
TCP isn't instant though.  RTO * retries can be hundreds of seconds if a link exhibits a high round-trip time.  Even a single lost packet could lead to a breakage for TCP here, whereas this code might accommodate such a thing for UDP.

I don't see that it makes sense to branch like this PR attempts to.  In this context, the only thing that is special about UDP+MRP as compared to TCP is that we have immediate access to retry parameters with MRP, whereas these are likely obfuscated for TCP.  But they still absolutely exist.

Reply:
https://github.com/project-chip/connectedhomeip/issues/12009, I also don't have good idea to get quick calculation for tcp margin, just create the ticket for follow-up. Temporarily I would use the 30s for margin for tcp

Reply:
@mrjerryjohns , 

The simplest thing would be to decouple from the exact nature of the underlying transport's reliability features.  We could instead say: transport attempts to be reliable.  Therefore, we should allow 1 strike before assuming the subscription is dead.  So given mMaxIntervalCeilingSeconds, fail at 2x mMaxIntervalCeilingSeconds.

As an aside, this PR's computation for maximum idle-state MRP timeout is not spec compliant.

Reply:
move the discussion to 12009 for the follow-up